### ISSUE TITLE
Exclude private draft frameworks from CASE API, except for valide SALT client sessions

### ISSUE THREAD

**Issue by jmarks** [opensalt/opensalt#629]
As a editor I want to prevent anyone who is not authenticated from access a private draft framework.

A private draft framework should only be accessible via the UI or API by a user who is authenticated with an active session and who is a Super User, Super Editor, or is an Org admin, Org editor or reviewer within that org, or is given individual rights to the framework.

**Comment by roverwolf** [comment]
I do not think this is a good idea.  This would mean that the framework could not be tested with an LMS or any other CASE client.

A framework that is marked as "Private Draft" does not get listed in the get all documents API call and so is only fetchable from the API if the URI had been obtained by someone with access to the framework in order to get it.

**Comment by roverwolf** [comment]
Please provide a use case where someone could legitimately obtain the URI but not be allowed to see what it points to.

**Comment by jmarks** [comment]
So this gets to the OAuth for the API question to model the LTI security model. I know that is a very slippery slope.

**Comment by jmarks** [comment]
Re: Usecase, 
1) The military has a secret agent competency framework and wants to take the secret agent test. 
2) A company has a set of skills related to a secret new project and wants to provide those to testing and curriculum partners who have secure environments
3) A state is developing new standards and wants to share them with partner states prior to publishing them.
4) Industry organizations, like the PMI, do not want to give away their frameworks but do want to be interoperable, and perhaps need a way to transition from their current proprietary model.
5) We want to protect the compendium and private crosswalks

Currently, any system can use Get /CFDocuments to see all frameworks on a server and then get each as a package. Also in the UI is there is an association to a private draft framework in a public framework you see that and can link to it as well. So it is very easy to find a private draft framework,

**Comment by brandonopened** [comment]
We are handling this in two ways:
1) internal server is behind firewall
2) utilizing the bidirectional association feature to allow “replaced from” association to be in the private draft framework so it doesn’t not show in the original framework. (Actually i should double check the case file on that)

**Comment by jmarks** [comment]
This may require adding OAuth access controls to API endpoints.

**Comment by jmarks** [comment]
Ideally there is a server configuration option to "Hide associations FROM private draft frameworks in the UI if the user does not have access to that private draft draft framework." In this way you can have a public framework that is crosswalked to "From" another framework and you will only see those "Related From" associations (e.g the associations that exist in a different package) unless logged in. This can be used to protect private frameworks and crosswalks resident on a common server.
